### PR TITLE
Fixes #30397 - Add DSL docs for tasks

### PR DESCRIPTION
**Pull request by ofedoren**

### CODE REVIEW

File: app/models/foreman_tasks/task.rb [hunk 1]
```diff
@@ -70,6 +70,17 @@ def check_permissions_after_save
            end)
     scope :for_action_types, (->(action_types) { where('foreman_tasks_tasks.label IN (?)', Array(action_types)) })
 
+    apipie :class, "A class representing #{model_name.human} object" do
+      name 'Task'
+      refs 'Task'
+      sections only: %w[all additional]
+      property :main_action, Class, desc: 'Returns the main action of the task, e.g. Actions::RemoteExecution::RunHostJob'
```
Comment (adamruzicka) [inline app/models/foreman_tasks/task.rb:77]: This should probably state that the return value isn't a class, but rather the instance

Comment (ofedoren) [reply app/models/foreman_tasks/task.rb]: Thanks, @adamruzicka, updated.

File: app/models/foreman_tasks/task.rb [hunk 2]
```diff
@@ -70,6 +70,17 @@ def check_permissions_after_save
            end)
     scope :for_action_types, (->(action_types) { where('foreman_tasks_tasks.label IN (?)', Array(action_types)) })
 
+    apipie :class, "A class representing #{model_name.human} object" do
+      name 'Task'
+      refs 'Task'
+      sections only: %w[all additional]
+      property :main_action, object_of: Class, desc: 'Returns the main action of the task, e.g. Actions::RemoteExecution::RunHostJob'
```
Comment (adamruzicka) [inline app/models/foreman_tasks/task.rb:77]: This says that the returned thing is an instance of class `Class`? If so, could we change it to `Dynflow::Action`?

Comment (ofedoren) [reply app/models/foreman_tasks/task.rb]: Makes more sense, thank you. Updated.

**Comment by adamruzicka** [comment]
Thank you @ofedoren !

**Pull request merged**